src/PhpSpreadsheet/Calculation/Logical.php

```diff
@@ -372,4 +372,40 @@ public static function IFNA($testValue = '', $napart = '')
 
         return self::statementIf(Functions::isNa($testValue), $napart, $testValue);
     }
+
+    /**
+     * IFS.
+     *
+     * Excel Function:
+     *         =IFS(testValue1;returnIfTrue1;testValue2;returnIfTrue2;...;testValue_n;returnIfTrue_n)
+     *
+     *         testValue1 ... testValue_n
+     *             Conditions to Evaluate
+     *         returnIfTrue1 ... returnIfTrue_n
+     *             Value returned if corresponding testValue (nth) was true
+     *
+     * @category Logical Functions
+     *
+     * @param mixed ...$arguments Statement arguments
+     *
+     * @return mixed|string The value of returnIfTrue_n, if testValue_n was true. #N/A error if none of testValues was true
+     */
+    public static function IFS(...$arguments)
+    {
+        if (count($arguments) % 2 != 0) {
+            return Functions::NA();
+        }
+
+        for ($i = 0; $i < count($arguments); $i += 2) {
+            $testValue = ($arguments[$i] === null) ? '' : Functions::flattenSingleValue($arguments[$i]);
+            $returnIfTrue = ($arguments[$i + 1] === null) ? '' : Functions::flattenSingleValue($arguments[$i + 1]);
+            $result = self::statementIf($testValue, $returnIfTrue, Functions::DUMMY());
+
+            if ($result !== Functions::DUMMY()) {
```

Comment:
Yes, I understand your concern that there is a possibility that the returned result may be a valid case of the string `'#Not Yet Implemented'`; unlikely, but not impossible.
I might be inclined to create something purely PHP detectable, that cannot exist in a spreadsheet, such as an exception. Maybe create a new EvaluateIfFalseException which can be returned as `$result` rather than thrown; and directly type tested rather than simply value tested.

Reply:
That's a good idea. Can I use just simple instance of generic Exception in that case: `$falseValueException = new Exception();`? Not sure if we need special Exception for that.

Reply:
Yes you could use a standard Exception, I was simply thinking about readabilityand providing a named exception where the name provided contextual meaning